contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5864,3 +5870,48 @@ is_bbf_db_ddladmin_operation(Oid namespaceId)
 
 	return false;
 }
+
+/*
+ * remove_db_name_in_schema - remove the db name and underscore at the beginning
+ * 	of the given string. It is used to unmap schema name in error messages.
+ *
+ * 	@param schema_name - char *
+ * 	@return - unmapped schema name char *
+ */
+static const char *
+remove_db_name_in_schema(const char *object_name)
+{
+	char		*cur_db_name;
+	char		**splited_object_name;
+	char		*schema_name = NULL;
+	char		*mutable_name;
+    size_t		db_name_len;
+    size_t		prefix_len;
+	size_t		schema_name_len;
+
+	mutable_name = pstrdup(object_name);
```

Comment:
is it necessary to copy this string? If not, we can remove it. I think  `split_object_name` will not modify the original string.